src/Sentry/Internal/MemoryMonitor.cs

```diff
@@ -0,0 +1,164 @@
+/*
+ * dotnet-gcdump needs .NET 6 or later:
+ * https://www.nuget.org/packages/dotnet-gcdump#supportedframeworks-body-tab
+ *
+ * Also `GC.GetGCMemoryInfo()` is not available in NetFX or NetStandard
+ */
+#if NET6_0_OR_GREATER && !(IOS || ANDROID)
+
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Internal;
+
+internal class MemoryMonitor : IDisposable
+{
+    private readonly long _totalMemory;
+
+    private readonly SentryOptions _options;
+    private readonly HeapDumpTrigger _dumpTrigger;
+
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+
+    private readonly Action _onCaptureDump; // Just for testing purposes
+    private readonly Action<string> _onDumpCollected;
+
+    public MemoryMonitor(SentryOptions options, Action<string> onDumpCollected, Action? onCaptureDump = null)
+    {
+        _options = options;
+        _dumpTrigger = options.HeapDumpTrigger
+                       ?? throw new ArgumentException("No heap dump trigger configured on the options", nameof(options));
+        _onDumpCollected = onDumpCollected;
+        _onCaptureDump = onCaptureDump ?? CaptureMemoryDump;
+
+        _totalMemory = GC.GetGCMemoryInfo().TotalAvailableMemoryBytes;
+
+        GarbageCollectionMonitor.Start(CheckMemoryUsage, _cancellationTokenSource.Token);
```

Comment:
We could and I even started playing around with it... but the earliest we can await the task is during MemoryMonitor.Dispose (after calling `token.Cancel()`). That's not ideal as we'd be logging the exception when shutting down the application (rather than when the exception actually occurred). 

Trying to fail early then, I went with your suggestion of using `Task.ContinueWith`... which means the error gets logged as soon as possible... albeit kind of behind the scenes in some task pool thread.